File: Top 100 Codes/02-Working with Numbers/02-Lowest Common Multiple (LCM)/LowestCommonMultipleLCM.py
# Function to calculate LCM directly
def calculate_lcm(a, b):
    greater = max(a, b)
    lcm = greater

    # Find the smallest multiple of the greater number that is divisible by the other number
    while lcm % a != 0 or lcm % b != 0:
        lcm += greater
    return lcm

# Prompt the user to enter two numbers
num1 = int(input("Enter the first number: "))
num2 = int(input("Enter the second number: "))

# Call the calculate_lcm function to compute the LCM
lcm = calculate_lcm(num1, num2)

# Output the result to the user
print(f"The LCM of {num1} and {num2} is: {lcm}")

File: Top 100 Codes/02-Working with Numbers/02-Lowest Common Multiple (LCM)/LowestCommonMultipleLCM.cpp
#include <iostream>
using namespace std;

// Function to calculate LCM directly using the formula: LCM(a, b) = (a * b) / HCF(a, b)
int calculateLCM(int a, int b) {
    int greater = max(a, b);
    int lcm = greater;

    // Find the smallest multiple of the greater number that is divisible by the other number
    while (lcm % a != 0 || lcm % b != 0) {
        lcm += greater;
    }
    return lcm;
}

int main() {
    int num1, num2;

    // Prompt the user to enter two numbers
    cout << "Enter two numbers to find their LCM: ";
    cin >> num1 >> num2;

    // Call the calculateLCM function to compute the LCM
    int lcm = calculateLCM(num1, num2);

    // Output the result to the user
    cout << "The LCM of " << num1 << " and " << num2 << " is: " << lcm << endl;

    return 0; // Indicate that the program ended successfully
}